documentation/src/docs/dev-guide.md

```diff
@@ -0,0 +1,203 @@
+
+
+
+
+## Create an application that writes to Pravega
+
+Let’s first create a pravega configuration reading from arguments:
+```java
+ParameterTool params = ParameterTool.fromArgs(args);
+PravegaConfig pravegaConfig = PravegaConfig
+        .fromParams(params)
+        .withDefaultScope("my_scope");
+```
+Then we need to initialize the Flink execution environment
+```java
+final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+```
+Create a datastream that gets input data by connecting to the socket
+```java
+DataStream<String> dataStream = env.socketTextStream(host, port);
+```
+A Pravega Stream may be used as a data sink within a Flink program using an instance of `io.pravega.connectors.flink.FlinkPravegaWriter`. We add an instance of the writer to the dataflow program:
+```java
+FlinkPravegaWriter<String> writer = FlinkPravegaWriter.<String>builder()
+        .withPravegaConfig(pravegaConfig)
+        .forStream(stream)
+        .withSerializationSchema(new SimpleStringSchema())
+        .build();
+dataStream.addSink(writer).name("Pravega Sink");
+```
+Then we execute the job within the Flink environment
+```java
+env.execute("PravegaWriter");
+```
+Executing the above lines should ensure we have created a PravegaWriter job
+
+## Create an application that reads from Pravega
+Creating a Word Count Reader is similar to Word Count Writer
+First create a pravega configuration reading from arguments:
+```java
+ParameterTool params = ParameterTool.fromArgs(args);
+PravegaConfig pravegaConfig = PravegaConfig
+        .fromParams(params)
+        .withDefaultScope("my_scope");
+```
+Initialize the Flink execution environment
+```java
+final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+```
+A Pravega Stream may be used as a data source within a Flink streaming program using an instance of `io.pravega.connectors.flink.FlinkPravegaReader`. The reader reads a given Pravega Stream (or multiple streams) as a DataStream
+```java
+FlinkPravegaReader<String> source = FlinkPravegaReader.<String>builder()
+        .withPravegaConfig(pravegaConfig)
+        .forStream(stream)
+        .withDeserializationSchema(new SimpleStringSchema())
+        .build();
+```
+Then create a datastream count each word over a 10 second time period
+```java
+DataStream<WordCount> dataStream = env.addSource(source).name("Pravega Stream")
+        .flatMap(new Tokenizer()) // The Tokenizer() splits the line into words, and emit streams of "WordCount(word, 1)"
+        .keyBy("word")
+        .timeWindow(Time.seconds(10))
+        .sum("count");
+```
+Create an output sink to print to stdout for verification
+```java
+dataStream.print();
+```
+Then we execute the job within the Flink environment
+```java
+env.execute("PravegaReader");
+```
+
```

Comment:
We may add an extra section or just some words here talking about `Run and Debug in local IDE`, this may help for local application development.